src/lib/components/api.js

```diff
@@ -0,0 +1,164 @@
+const githubOwner = "SSWConsulting";
+const githubRepo = "SSW.Rules.Content";
+const apiBaseUrl = "https://api.github.com/graphql";
+
+export async function fetchPullRequests(numberOfRules, author, token) {
+  const pullRequests = await requestPullRequests(
+    numberOfRules,
+    author,
+    token
+  );
+  pullRequests.sort((a, b) => new Date(b.mergedAt) - new Date(a.mergedAt));
+  return pullRequests;
+}
+
+export async function fetchFileContents(filesToRetrieve, numberOfRules, token) {
+  var fileContents = await requestMultipleFileContents(
+    filesToRetrieve,
+    numberOfRules,
+    token
+  );
+
+  var counter = 0;
+
+  while (
+    fileContents.filter((x) => x !== null).length < numberOfRules
```

Comment:
The purpose of this filter is fairly unclear. I am not sure what it is doing?

Perhaps encapsulating this while loop logic in a function might make it clearer